### PR TITLE
Granite and GraniteMoE models.

### PR DESCRIPTION
**Pull request by janimo**
Rouge scores are good but logit values are very off.

### CODE REVIEW

**Comment by merrymercy** [comment]
let us know when it is ready to be merged.

**Comment by janimo** [comment]
I still need to debug why the logits in the ONLY_RUN test go so much over the thresholds (e.g. 60 instead of << 1 like in others) even if the rogue scores match. You can merge if you wish and I can continue debugging or I'll update the PR when I find it. I am not sure whether implementation details should cause such large differences.

**Comment by frreiss** [comment]
@janimo sorry, just noticed that this PR exists. I submitted a similar PR yesterday with Granite integration code that came out of our work on a NeurIPs booth demo. My PR is at https://github.com/sgl-project/sglang/pull/1818, and it appears to have been merged already.

@merrymercy would you mind also including @janimo in the credits for Granite support on the release notes of the next version?

**Comment by janimo** [comment]
@frreiss no problem. Apparently (I haven't tested, I had no GPU access lately) my missing the need for a granite specific chat template could have been the cause of the large logit value differences. Possibly the MoE model could pass now too.

**Comment by merrymercy** [comment]
Sure. I will do that and close this PR for now.
